### PR TITLE
Enochtsang/bms can parser

### PR DESCRIPTION
**Pull request by enochtsang**
March 4th: "If I work really hard, I can finish CCS in 2 weeks."

Doing this PR: "This is alot harder than I thought..."

Deja Vu @JordanHeinrichs?

### CODE REVIEW

File: EpsilonCCS/Inc/CanParsers/BmsCanParser.h [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#pragma once
+
+#include "stm32f4xx.h"
+
+void parseBmuCanMessage(uint32_t stdId, uint8_t data[8]);
```
Comment (JordanHeinrichs) [inline EpsilonCCS/Inc/CanParsers/BmsCanParser.h]: I will just leave you with the great Linus Torvalds on passing arrays into functions. https://lkml.org/lkml/2015/9/3/428

File: EpsilonCCS/Src/CanParsers/BmsCanParser.c [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+void parseStartupInfo(uint8_t data[8])
+{
+
+    batteryData.bmsRelayStatus = data[0];
+
+    batteryData.populatedCells = data[1];
+
+    uint16_t input12voltInt = // Units 0.1 V
+        (data[2] << 0) +
```
Comment (JordanHeinrichs) [inline EpsilonCCS/Src/CanParsers/BmsCanParser.c]: Change all instances of + to `|`. This is an OR operation so best to use the OR operator. While using a + is basically the same it is not exactly and can introduce hard to find bugs (mostly related to int overflow and/or negatives).

File: EpsilonCCS/Inc/CanParsers/BmsCanParser.h (hunk 1 above)
Comment (enochtsang) [inline EpsilonCCS/Inc/CanParsers/BmsCanParser.h]: I feel ashamed.

**Comment by enochtsang** [comment]
I've added my AuxBms code as well.

File: EpsilonCCS/Inc/CanParsers/BmsCanParser.h (hunk 1 above)
Comment (JordanHeinrichs) [inline EpsilonCCS/Inc/CanParsers/BmsCanParser.h]: Yeah, the only thing I really prefer about it though is it gets rid of the magic number 8.

File: EpsilonCCS/Src/CanParsers/AuxBmsCanParser.c [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+#include "AuxBmsCanParser.h"
+
+#include "BatteryData.h"
+
+extern struct BatteryData batteryData;
+
+#define AUX_BMS_HEARTBEAT_ID (0x720)
+#define AUX_BMS_INFO_ID (0x721)
+
+#define PRECHARGE_STATE_MASK (0x00) // First 3 bits
```
Comment (bill-luu) [inline EpsilonCCS/Src/CanParsers/AuxBmsCanParser.c]: Or would it be 0x70? Since line 13 has 0xF8 for the last 5 bits?

**Comment by enochtsang** [comment]
Replaced by https://github.com/UCSolarCarTeam/Epsilon-Embedded-Software/pull/18

**Pull request closed without merging**